### ISSUE TITLE
Pipeline Tool file linking

### ISSUE THREAD

**Issue by KakCAT** [MonoGame/MonoGame#5658]
Hello,

I've been patching the Pipeline tool to accept linked files. 

Some notes:

* The patch includes @vchelaru 's PR to support links in MGCB  https://github.com/MonoGame/MonoGame/pull/5574 with the addition of links to 'copied' files, which were missing (I'm so lame at git that I had to copy them manually to my repository)
* When deleting linked files, "LINKED TO" appears in the delete list to show they're linked files. There is surely a better way.
* To mimic Visual Studio links, when using "Alt" key to drag/drop files, no confirmation is required and "Link" is selected, as well as applyToAll.
* To mimic VS, when adding links that are not included in the project (because they already are included) a message box is shown. (In VS it appears in the log, but I didn't find a log to include it, and it was very confusing when files were not added without apparent reason). The message does not appear when copying (unless I messed up)

Unfortunately using git when other human users are around is too much for an old git like me (pun intended) and I've set up a project and uploaded all the changes there (that's the most I can do in git without spending an hour looking at documentation).
Here it is:  https://github.com/MonoGame/MonoGame/compare/develop...KakCAT:pipeline_tool_link_patch   (I don't even know if this is the link I should be posting XD ) 

It also mixes several different features in a single PR, which shouldn't be done, and some variable names might not be congruent with the coding style of the project. Not even mentioning tabs and spaces and formatting :)

Finally, I have no idea at all of the Pipeline tool, so some pieces of code are more a dirty hack than a nice piece for a PR. I just hope the repository is useful to somebody with more Pipeline tool knowledge to integrate it. On the bright side I've overcommented some parts to give the 'integrator' a clue of what I'm doing :)

Missing features I haven't been able to figure out:
* When deleting a linked file, the tree is not updated (I spent hours trying to figure why, so I left it blank). When deleting folders it is OK.  It is very annoying (do you have any clue, @cra0zy ? )
* Icons should have a "link" icon when linking.
* To mimic Visual Studio links, deleting a bunch of linked files should show no delete confirmation.
* The linking system is a bit sensitive when it founds '\\' instead of '/', which shouldn't appear unless you edit the mgcb manually. I suppose the fix to this problem should be when loading the file from disk, so I decided not to touch that.
* Renaming linked assets do not work (it's easily doable but I don't know if it should be done. VS won't let you)
* Some bits will only work in windows (no idea of xwt, sorry)

I've been testing this for a few days with 2 of my projects without problems, but maybe my workflow doesn't show them. At least I think it's a good starting port to support file linking referred in #5109

**Comment by cra0zy** [comment]
Once #5574 gets merged I'll add proper Pipeline Tool support for the linked files. For now this just feels like another duplicate of #5109

**Comment by KakCAT** [comment]
Actually this contains a link to a Pipeline Tool version where linking files works. I suppose you read the "I don't know how to do that" ranting part where I mentioned you ;)

The message is lengthy because I can't do a PR and I tried to explain what I did, but I think it's a good starting point after #5574 ends.

**Comment by cra0zy** [comment]
I've read it, and your patch does not solve all the problems. Actually to be precise it breaks more stuff than it solves...

**Comment by KakCAT** [comment]
Sorry then. You can close it if you want.

**Comment by Apostolique** [comment]
I believe this is good to close.
